symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/pagination/CursorBasedPaginatedService.java

```diff
@@ -0,0 +1,76 @@
+package com.symphony.bdk.core.service.pagination;
+
+import com.symphony.bdk.core.service.pagination.model.CursorPaginatedPayload;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiRuntimeException;
+
+import org.apiguardian.api.API;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+@API(status = API.Status.INTERNAL)
+public class CursorBasedPaginatedService<T> extends PaginatedService {
+  private CursorBasedPaginatedApi<T> paginatedApi;
+
+  /**
+   * @param paginatedApi the paginated api to be called
+   * @param chunkSize    the max number of items to be retrieved in one call
+   * @param maxSize      the total max number of items to be retrieved
+   */
+  public CursorBasedPaginatedService(CursorBasedPaginatedApi<T> paginatedApi, Integer chunkSize, Integer maxSize) {
+    super(chunkSize, maxSize);
+    this.paginatedApi = paginatedApi;
+  }
+
+  /**
+   * Returns an stream of elements with lazy fetching.
+   *
+   * @return a {@link Stream} which lazily makes calls to the {@link #paginatedApi}.
+   */
+  public Stream<T> stream() {
```

Comment:
`@Override` missing?

Reply:
No, the method is not actually defined in the base class and I don't see any value since we directly instantiate the child classes.